pilot/pkg/proxy/envoy/v2/mesh.go

```diff
@@ -0,0 +1,990 @@
+// configuration element is updated independently of all other
+// configuration elements within this pilot.
+const (
+	_                          = iota
+	ConfigAdd ConfigChangeType = iota
+	ConfigUpdate
+	ConfigDelete
+)
+
+// Enumerated constants for SocketProtocol that's coupled with
+// the internal implementation (Envoy lbEndoi
+const (
+	_                                = iota
+	SocketProtocolTCP SocketProtocol = iota
+	SocketProtocolUDP
+)
+
+// Enumerated constants for DestinationRuleType based on how route.DestinationRule.Name
+// should be interpreted.
+// TODO: Move DestinationRuleType to github.com/istio.io/api
+const (
+	_ = iota
+	// DestinationRuleService is a type of destination rule where the
+	// rule name is an FQDN of the service and resulting Subsets ought
+	// to be further scoped to this FQDN.
+	DestinationRuleService DestinationRuleType = iota
+
+	// DestinationRuleName is a type of destination rule where
+	// the rule name is the short name of the service and the
+	// resulting subset ought to be further scoped to only those
+	// Endpoints whose short service name match this short name.
+	DestinationRuleName
+
+	// DestinationRuleIP is a type of destination rule where
+	// the rule name is a specific IP and the resulting subset
+	// ought to be further scoped to this Endpoint's IP to determine
+	// the Subset.
+	DestinationRuleIP
+
+	// DestinationRuleWildcard is a type of destination rule where
+	// the rule name is a wild card domain name and the
+	// resulting subset ought to be further scoped to only those
+	// Endpoints whose domains match this wild card domain.
+	DestinationRuleWildcard
+
+	// DestinationRuleCIDR is a type of destination rule where
+	// the rule name is a CIDR and the resulting subset
+	// ought to be further scoped to only those Endpoints whose
+	// IPs that match this CIDR.
+	DestinationRuleCIDR
+)
+
+var (
+	// prohibitedAttrSet contains labels that should not be supplied
+	// to NewEndpoint(.... labels)
+	prohibitedAttrSet map[string]bool
+)
+
+// DestinationAttribute encapsulates enums for key Istio attribute names used in Subsets
+// and Endpoints
+type DestinationAttribute string
+
+// Mesh is a platform independent abstraction used by Controllers
+// for maintaining a list of service endpoints used by this Pilot.
+//
+// Controllers under https://github.com/istio/istio/tree/master/pilot/platform
+// update the list of service endpoints in this Mesh with those
+// available in a platform specific service registry.
+//
+// Under the hoods, the Mesh implements the necessary logic required for endpoint
+// discovery based on routing rules and endpoint subsets.
+// See https://github.com/istio/api/search?q=in%3Afile+"message+Subset"+language%3Aproto
+// This logic includes comparing the updated list provided by Controllers with
+// what this view holds and accordingly updating internal structures used for
+// endpoint discovery.
+type Mesh struct {
```

Comment:
Nit. We already have a proto by name Mesh.. Its all over the config generation code. That needs to be resolved one way or another.

Reply:
Could you provide a reference? just in case  the other proto needs renaming

Reply:
I would suggest you rename this file to something else, as the one in proto is heavily tied to our docs (and rightfully, mesh there implies mesh wide configuration).

Reply:
I don't see a problem atm. go allows for aliases. Given that the context for the proto is config and the context for this file is the data model, it should be fine.